File: engine/src/main/java/de/hanno/hpengine/engine/graphics/renderer/rendertarget/ColorAttachmentDefinition.java
package de.hanno.hpengine.engine.graphics.renderer.rendertarget;

import org.lwjgl.opengl.GL11;

public class ColorAttachmentDefinition {
    int internalFormat = GL11.GL_RGB;
    int textureFilter = GL11.GL_LINEAR;

    public ColorAttachmentDefinition() {}

    public ColorAttachmentDefinition setInternalFormat(int internalFormat) {
        this.internalFormat = internalFormat;
        return this;
    }
    public ColorAttachmentDefinition setTextureFilter(int textureFilter) {
        this.textureFilter = textureFilter;
        return this;
    }
}

File: engine/src/main/java/de/hanno/hpengine/engine/graphics/renderer/rendertarget/ColorAttachmentDefinition.kt
package de.hanno.hpengine.engine.graphics.renderer.rendertarget

import de.hanno.hpengine.engine.graphics.renderer.constants.TextureFilterConfig
import org.lwjgl.opengl.GL11

data class ColorAttachmentDefinition @JvmOverloads constructor(val name: String, val internalFormat: Int = GL11.GL_RGB, var textureFilter: TextureFilterConfig = TextureFilterConfig()) {
    fun setInternalFormat(internalFormat: Int): ColorAttachmentDefinition {
        return copy(internalFormat = internalFormat) // TODO: Use this pattern everywhere
    }

    fun setTextureFilter(textureFilter: TextureFilterConfig): ColorAttachmentDefinition {
        this.textureFilter = textureFilter
        return this
    }
}

data class ColorAttachmentDefinitions @JvmOverloads constructor(val names: Array<String>, var internalFormat: Int = GL11.GL_RGB, var textureFilter: TextureFilterConfig = TextureFilterConfig())
